content/post/52-go-ipfs-0.4.18.md

```diff
@@ -0,0 +1,283 @@
+installing go-ipfs and visiting http://localhost:5001/webui. It deserves its own release note - oh look, it [got
+one](./51-js-ipfs-0-33/#web-ui-2-0)! Here's a peek:
+
+![Screenshot of the status page](https://raw.githubusercontent.com/ipfs-shipyard/ipfs-webui/master/docs/screenshots/ipfs-webui-status.png)
+
+| Files | Explore | Peers | Settings |
+|-------|---------|-------|----------|
+| ![Screenshot of the file browser page](https://raw.githubusercontent.com/ipfs-shipyard/ipfs-webui/master/docs/screenshots/ipfs-webui-files.png) | ![Screenshot of the IPLD explorer page](https://raw.githubusercontent.com/ipfs-shipyard/ipfs-webui/master/docs/screenshots/ipfs-webui-explore.png) | ![Screenshot of the swarm peers map](https://raw.githubusercontent.com/ipfs-shipyard/ipfs-webui/master/docs/screenshots/ipfs-webui-peers.png) | ![Screenshot of the settings page](https://raw.githubusercontent.com/ipfs-shipyard/ipfs-webui/master/docs/screenshots/ipfs-webui-settings.png) |
+
+
+Kudos and thanks to the webui team!
+
+## Performance
+
+This release includes some significant performance improvements, both in terms
+of resource utilization and speed. This section will go into some technical
+details so feel free to skip it if you're just looking for shiny new features.
+
+### Resource Utilization
+
+In this release, we've (a) fixed a slow memory leak in libp2p and (b)
+significantly reduced the allocation load. Together, these should improve both
+memory and CPU usage.
+
+#### Datastructures
+
+We've changed two of our most frequently used datastructures, CIDs and
+Multiaddrs, to reduce allocation load.
+
+First, we now store CIDs *encode* as strings, instead of decoded in structs
+(behind pointers). In addition to being more compact, our `Cid` type is now a
+valid `map` key so we no longer have to encode CIDs every time we want to use
+them in a map/set. Allocations when inserting CIDs into maps/sets was showing up
+as a significant source of allocations under heavy load so this change should
+improve memory usage.
+
+Second, we've changed many of our multiaddr parsing/processing/formatting
+functions to allocate less. Much of our DHT related-work includes processing
+multiaddrs so this should reduce CPU utilization when heavily using the DHT.
+
+#### Streams and Yamux
+
+Streams have always plagued us in terms of memory utilization. This was
+partially solved by introducing the connection manager, keeping our maximum
+connection count to a reasonable number but they're still a major memory sink.
+
+This release sees two improvements on this front:
+
+1. A memory [leak in identify](https://github.com/libp2p/go-libp2p/issues/419)
+   has been fixed. This was slowly causing us to leak connections (locking up
+   the memory used by the connections' streams).
+2. Yamux streams now use a buffer-pool backed, auto shrinking read buffer.
+   Before, this read buffer would grow to its maximum size (a few megabytes) and
```

Comment:
> Before, this read buffer would grow to its maximum size (a few megabytes) and
   never shrink, but **now** these buffers shrink as they're emptied **and free up space efficiently**.